packages/components/src/components/Address/Address.tsx

```diff
@@ -1,16 +1,67 @@
-import { Input } from '@chakra-ui/react';
-import React from 'react';
+import {
+  FormControl,
+  FormErrorMessage,
+  Input,
+  InputGroup,
+  InputRightElement,
+} from '@chakra-ui/react';
+import { CopyIcon, CheckIcon } from '@chakra-ui/icons';
+import React, { useState } from 'react';
 
 export interface AddressProps {
   /**
    * The address to display
    */
   value: string;
+  /**
+   * Wheter the address can be copied or not
+   */
+  copiable?: boolean;
+}
+
+interface EventTarget {
+  value: string;
+}
+
+interface SyntheticEvent {
+  currentTarget: EventTarget;
 }
 
 /**
  * A component to display an address
  */
-export const Address: React.FC<AddressProps> = ({ value }) => {
-  return <Input value={value} />;
+export const Address: React.FC<AddressProps> = ({ value, copiable = false }) => {
+  const [error, setError] = useState<null | string>(null);
+  const [copied, setCopied] = useState<boolean>(false);
+
+  const handleClick = async (event: SyntheticEvent): Promise<void> => {
+    const value = event.currentTarget.value as string;
+
+    setError(null);
+    setCopied(false);
```

Comment:
I don't think this is necessary since it `copied` will be `true` almost instantly anyways later on in this very function.

What I suggest is that we set a timeout or something that sets `copied` back to `false` after 2 or 3 seconds so that the user can see the 'copy' icon after 2-3 seconds if they want to copy it again.